### PR TITLE
Checking-in Go generated code?

### PR DESCRIPTION
**Pull request by ozkatz**
Feel free to close (I don't have a strong opinion on this), but it might play nicely with #7616.

### Upsides

- Makes lakeFS `go get`-able
- Reduces build time by an insignificant amount of time (~2 seconds?)

### Downsides

- Adds an insignificant amount of space (<1MB) to the repo?

---
Happy to hear other up/downsides and opinions.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -276,6 +274,12 @@ validate-mockgen: gen-code
 	git diff --quiet -- pkg/kv/mock/store.go || (echo "Modification verification failed! pkg/kv/mock/store.go"; false)
 	git diff --quiet -- pkg/pyramid/mock/pyramid.go || (echo "Modification verification failed! pkg/pyramid/mock/pyramid.go"; false)
 
+.PHONY: validate-apigen
+validate-apigen: gen-api
```
Comment (N-o-Z) [inline Makefile:278]: Might want to add this also to checks-validator target

File: pkg/api/apigen/lakefs.gen.go
Comment (N-o-Z) [inline pkg/api/apigen/lakefs.gen.go:1]: Should add the generated files to the linguist ignore under .gitattributes

File: Makefile (hunk 1 above)
Comment (ozkatz) [reply Makefile:278]: Done

File: pkg/api/apigen/lakefs.gen.go
Comment (ozkatz) [reply pkg/api/apigen/lakefs.gen.go:1]: Done

**Review by arielshaqed — CHANGES_REQUESTED** [review]
The code is of course good.  Generated code includes both client and server sides.  I believe that we had two reasons not to release so far.  I am not sure where we stand on either one right now.

1. The server code has some additions to the default templating.  This is pretty internal stuff (although I did not see anything lakeFS-specific): if we release the generated server code then we need to offer SemVer on it as well.
1. The client and server codes are mixed up in the same package and even in the same file.  Figuring out which is which is _very_ confusing.

## A way ahead

I do not believe that we should publish any generated Go server code.  Given the Go source "publication" model, I do understand why we would want to publish Go client code like this.  I would be very happy to generate _separate_ client and server packages, and to check in and publish _only_ the client package.

Also we should consider whether we want to delay such publication until after #7616, and go directly to a separate module.

**Comment by ozkatz** [comment]
OK. closing this PR for now.

**Pull request closed without merging**